signal/signal.go

```diff
@@ -95,16 +99,22 @@ func (c *Interceptor) mainInterruptHandler() {
 		select {
 		case signal := <-c.interruptChannel:
 			log.Infof("Received %v", signal)
+			// Adding exit status code 0 as this will quit after graceful shutdown
+			c.setExitCode(0)
 			shutdown()
 
 		case <-c.shutdownRequestChannel:
 			log.Infof("Received shutdown request.")
+			// Adding exit status code 0 as this will quit after graceful shutdown
+			c.setExitCode(1)
 			shutdown()
 
 		case <-c.quit:
 			log.Infof("Gracefully shutting down.")
 			close(c.shutdownChannel)
 			signal.Stop(c.interruptChannel)
+			// Adding exit status code 0 as this will quit after graceful shutdown
+			c.setExitCode(0)
```

Comment:
Do we need to set an exit code here? I think we can only hit this case if we've called `shutdown` in one of the other cases? 

We can't send multiple values into the channel otherwise we'll end up blocked. I think it's worth adding a comment above `<-c.quit` that we only reach this case once shutdown has been called.